invenio_app_rdm/theme/assets/js/invenio_app_rdm/search/SearchFacets/SearchFacets.js

```diff
@@ -0,0 +1,58 @@
+// This file is part of InvenioRDM
+// Copyright (C) 2019 CERN.
+// Copyright (C) 2019 Northwestern University.
+//
+// Invenio App RDM is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component, Fragment } from "react";
+import { Checkbox, Label, List } from "semantic-ui-react";
+import { BucketAggregation } from "react-searchkit/esm";
+import { config } from "../config";
+
+export class SearchFacets extends Component {
+  _renderValueElement = (
+    bucket,
+    isSelected,
+    onFilterClicked,
+    getChildAggCmps
+  ) => {
+    const childAggCmps = getChildAggCmps(bucket);
+    const key = bucket.key_as_string ? bucket.key_as_string : bucket.key;
+    return (
+      <List.Item key={bucket.key}>
+        <List.Content floated="right">
+          <label>{bucket.doc_count}</label>
+        </List.Content>
+        <List.Content>
+          <Checkbox
+            label={<label>{key}</label>}
+            value={key}
+            onClick={() => onFilterClicked(key)}
+            checked={isSelected}
+          />
+          {childAggCmps}
+        </List.Content>
+      </List.Item>
+    );
+  };
```

Comment:
Arrow functions are actual the solution on `this-bind` issue. The only caveat is that they are not inherited but in React you should not inherit other components as it can lead to some nasty errors. Did I get correctly your question?

Reply:
Yes. That's what I thought about the anonymous function. But I didn't know about how inheritance was not recommended in React. Thanks.